flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,472 @@
+     */
+    private static final int MAX_NESTED_LEVEL = 15;
+
+    private static final Map<Schema.Type, StandardSQLTypeName> AVRO_TYPES_TO_BQ_TYPES;
+    private static final Map<String, StandardSQLTypeName> LOGICAL_AVRO_TYPES_TO_BQ_TYPES;
+
+    // ----------- Initialize Maps between Avro Schema to BigQuery schema -------------
+    static {
+        /*
+         * Map Avro Schema Type to StandardSQLTypeName which converts AvroSchema
+         * Primitive Type to StandardSQLTypeName.
+         * AVRO_TYPES_TO_BQ_TYPES: containing mapping from Primitive Avro Schema Type to StandardSQLTypeName.
+         */
+        AVRO_TYPES_TO_BQ_TYPES = new EnumMap<>(Schema.Type.class);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.INT, StandardSQLTypeName.INT64);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.FIXED, StandardSQLTypeName.BYTES);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.LONG, StandardSQLTypeName.INT64);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.FLOAT, StandardSQLTypeName.FLOAT64);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.DOUBLE, StandardSQLTypeName.FLOAT64);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.STRING, StandardSQLTypeName.STRING);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.BOOLEAN, StandardSQLTypeName.BOOL);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.ENUM, StandardSQLTypeName.STRING);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.BYTES, StandardSQLTypeName.BYTES);
+
+        /*
+         * Map Logical Avro Schema Type to StandardSQLTypeName Type, which converts
+         * AvroSchema Logical Type to StandardSQLTypeName.
+         * LOGICAL_AVRO_TYPES_TO_BQ_TYPES: Map containing mapping from Logical Avro Schema Type to StandardSQLTypeName.
+         */
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES = new HashMap<>();
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(LogicalTypes.date().getName(), StandardSQLTypeName.DATE);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMicros().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMillis().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.uuid().getName(), StandardSQLTypeName.STRING);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMillis().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMicros().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMillis().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMicros().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("geography_wkt", StandardSQLTypeName.GEOGRAPHY);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("Json", StandardSQLTypeName.JSON);
```

Comment:
We should create constants for geography and json string type names, and re-use wherever needed